test/GithubApi.Gist.Fetch.test.js

```diff
@@ -0,0 +1,114 @@
+const isomorphic = require('isomorphic-fetch');
+const chai = require('chai');
+const statusCode = require('http-status-codes');
+
+chai.use(require('chai-subset'));
+
+const { expect } = chai;
+
+const jsCode = `
+function wait(method, time) {
+  return new Promise((resolve) => {
+    setTimeout(resolve(method()), time);
+  });
+}
+`;
+
+const urlBase = 'https://api.github.com';
+const defaultHeaders = {
+  Authorization: `token ${process.env.ACCESS_TOKEN}`
+};
+let gist;
+
+describe('Given a github user', () => {
+  describe('when create a gist', () => {
+    const createGist = {
+      description: 'this is an example about promise',
+      public: true,
+      files: {
+        'promise.js': {
+          content: jsCode
+        }
+      }
+    };
+
+    let newGistQuery;
+    let statusNewGistQuery;
+
+    before(() => {
+      const parameters = {
+        method: 'POST',
+        body: JSON.stringify(createGist),
+        headers: defaultHeaders
+      };
+
+      newGistQuery = isomorphic(`${urlBase}/gists`, parameters)
+        .then((response) => {
+          statusNewGistQuery = response.status;
+          return response.json();
+        })
+        .then((body) => {
+          gist = body;
+          return body;
+        });
+
+      return newGistQuery;
+    });
+
+    it('then a new gist should be created', () => {
+      newGistQuery
+        .then(() => {
+          expect(statusCode.CREATED).to.equal(statusNewGistQuery);
```

Comment:
to keep the consistency, change this to:
`expect(statusNewGistQuery).to.equal(statusCode.CREATED);`
to have always the expected value as the second part